src/main/java/bdv/fx/viewer/ViewerPanelFX.java

```diff
@@ -390,21 +390,24 @@ public void requestRepaint(final long[] min, final long[] max)
 	public synchronized void transformChanged(final AffineTransform3D transform)
 	{
 		viewerTransform.set(transform);
-		state.setViewerTransform(transform);
+		synchronized (state)
+		{
+		    state.setViewerTransform(transform);
+		    renderingModeController.transformChanged();
+		}
 		for (final TransformListener<AffineTransform3D> l : transformListeners)
 			l.transformChanged(viewerTransform);
-		renderingModeController.transformChanged();
 		requestRepaint();
 	}
 
 	/**
-	 * Get a copy of the current {@link ViewerState}.
+	 * Get the current {@link ViewerState}.
 	 *
-	 * @return a copy of the current {@link ViewerState}.
+	 * @return the current {@link ViewerState}.
 	 */
 	public ViewerState getState()
 	{
-		return state.copy();
+		return state;
```

Comment:
To be able synchronize on viewer state in `RenderUnit.Paintable.paint()`, I changed this method to return the actual object and not a copy. I also changed other places that used this method and relied on a copy of the state to use synchronization instead.

Reply:
While this comes at the risk of callers modifying the state directly, it is probably fine to not a return a copy here. Once (or if) we return to multi-resolution renderer w/region-of-interest rendering we should think about reverthing this change.